### ISSUE TITLE
Invalid URL variable schema saved in requests

### ISSUE THREAD

**Issue by wasaylor** [a85/POSTMan-Chrome-Extension#3212]
## App Details:

```
Postman for Mac
Version 5.0.0 (5.0.0)
OS X 10.11.6 / x64
```

## Issue Report:
 1. Did you encounter this recently, or has this bug always been there:
Just started using Postman again yesterday (fresh install,) converted a v1 collection to v2 with the transform-collection app
 2. Expected behaviour:
When creating new, or modifying existing requests, the URL variables are saved with invalid schema. Postman seems to be able to handle this when sending the request manually, but the runner is ignoring the URL variables altogether and sends a bad request.

Example URL `https://example.com/resource/:id` with `:id` set to `1001` is sent as `https://example.com/resource/`.

After exporting the collection, the JSON for the request URL variable shows `"key": "id"` When I manually change that to `"id": "id"`, then import the collection, runner recognizes the variable. I followed the docs here: https://schema.getpostman.com/json/collection/v2.0.0/docs/index.html

 3. Console logs (http://blog.getpostman.com/2014/01/27/enabling-chrome-developer-tools-inside-postman/ for the Chrome App, View->Toggle Dev Tools for the Mac app):
(Doesn't seem to be logging when creating or saving requests)
(Doesn't seem to be logging when running the collection)

 4. Screenshots (if applicable)
Here's the JSON schema of the specific request (`item`):
```
{
	"name": "{{api_base}}/test/:id",
	"request": {
		"url": {
			"raw": "{{api_base}}/test/:id",
			"host": [
				"{{api_base}}"
			],
			"path": [
				"test",
				":id"
			],
			"query": [],
			"variable": [
				{
					"key": "id",
					"value": "1001"
				}
			]
		},
		"method": "GET",
		"header": [],
		"body": {},
		"description": ""
	},
	"response": []
}
```
(Obviously a fake request, but the issue still occurs)

1. Started with a v1 Postman collection, use the collection transform to convert to 2.0.0
2. Created/modified any request in the collection with URL variables
3. Observe collection runner not sending said requests correctly
4. Export collection and observe incorrect JSON schema
5. Fix the JSON schema and import the collection
6. Collection runner now sends the request properly

**Comment by kunagpal** [comment]
@wasaylor A fix for this bug was added in https://github.com/postmanlabs/postman-collection-transformer/pull/60, and will be rolled out with the next release. Thanks for reporting this 😄

**Comment by wasaylor** [comment]
@kunagpal Just for clarity, this happens when I create new requests in Postman v5 itself and use the runner - so after I've used the transformer. Just want to confirm you understand because it looks like you've only done a fix on the transformer itself

**Comment by kunagpal** [comment]
@wasaylor The Postman app depends on the transformer to perform various conversions. The next release I was referring to is the Postman app release, and not just the transformer release :)

**Comment by cebor** [comment]
I have the same issue with 5.0.2

Runner not usable for new or edited requests.

**Comment by wasaylor** [comment]
@kunagpal Hi, still happens in 5.0.2 - *also* postman looks like its duplicating path variables when saving. I export/edit/import my collection a lot, and now my path variables for each request look like this:

```
"variable": [
	{
		"id": "id",
		"value": "{{REDACTED}}"
	},
	{
		"id": "id",
		"value": "{{REDACTED}}"
	},
	{
		"value": "{{REDACTED}}",
		"id": "id"
	},
	{
		"value": "{{REDACTED}}",
		"id": "id"
	},
	{
		"value": "{{REDACTED}}",
		"id": "id"
	}
]
```

**Comment by cebor** [comment]
Seems to be fixed in 5.1.2

**Comment by sdnts** [comment]
Awesome, closing this.
